### PR TITLE
Fix std::fmt format spec: named args are allowed with "$" syntax

### PR DESCRIPTION
**Pull request by birkenfeld**

### CODE REVIEW

File: src/libcollections/fmt.rs [hunk 1]
```diff
@@ -456,6 +456,10 @@
 //! // Hello {next arg (x)} is {arg 2 (0.01) with precision
 //! //                          specified in its predecessor (5)}
 //! println!("Hello {} is {2:.*}",   "x", 5, 0.01);
+//!
+//! // Hello {next arg (x)} is {arg "number" (0.01) with precision specified
+//!                             in arg "prec" (5)}
```
Comment (GuillaumeGomez) [inline src/libcollections/fmt.rs]: Missing "//" at the beginning of the line.

**Comment by GuillaumeGomez** [comment]
Except for the nit, everything's good. Thanks for the PR. Ping me once it's fixed.

**Comment by birkenfeld** [comment]
@GuillaumeGomez thanks, fixed.

File: src/libcollections/fmt.rs [hunk 2]
```diff
@@ -333,7 +333,7 @@
 //! precision := count | '*'
 //! type := identifier | ''
 //! count := parameter | integer
-//! parameter := integer '$'
+//! parameter := argument '$'
```
Comment (steveklabnik) [inline src/libcollections/fmt.rs]: wouldn't we want this to be integer _or_ argument?

File: src/libcollections/fmt.rs [hunk 3]
```diff
@@ -402,12 +402,12 @@
 //! // All of these print "Hello x    !"
 //! println!("Hello {:5}!", "x");
 //! println!("Hello {:1$}!", "x", 5);
-//! println!("Hello {1:0$}!", 5, "x");
+//! println!("Hello {:width$}!", "x", width = 5);
```
Comment (steveklabnik) [inline src/libcollections/fmt.rs]: probably worth leaving both examples

File: src/libcollections/fmt.rs (hunk 2 above)
Comment (birkenfeld) [inline src/libcollections/fmt.rs]: `argument` is defined above as `integer | identifier`.

File: src/libcollections/fmt.rs (hunk 3 above)
Comment (birkenfeld) [inline src/libcollections/fmt.rs]: can do!

File: src/libcollections/fmt.rs (hunk 2 above)
Comment (steveklabnik) [inline src/libcollections/fmt.rs]: Ah! I missed that :+1:

**Comment by bluss** [comment]
:star2: Nice to get this into the docs, then it also gets into rust code out there eventually

**Comment by alexcrichton** [comment]
@bors: retry

* #33316

**Pull request merged**